.github/workflows/pypi_publish.yml

```diff
@@ -0,0 +1,310 @@
+name: Publish to PyPi
+
+on:
+  workflow_dispatch:
+    inputs:
+      version:
+        description: "Version to publish (leave empty for release version)"
+        required: false
+        type: string
+      test_publish:
+        description: "Publish to Test-PyPi only"
+        required: false
+        default: true
+        type: boolean
+
+permissions:
+  contents: read
+
+jobs:
+  linux:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        platform:
+          - target: x86_64
+          - target: i686
+          - target: aarch64
+          - target: armv7l
+          - target: ppc64le
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@08c6903cd8c0fde910a37f88322edcfb5dd907a8 # v5.0.0
+
+      - name: Install Rust
+        uses: dtolnay/rust-toolchain@0f44b27771c32bda9f458f75a1e241b09791b331
+        with:
+          toolchain: stable #1.91.1
+
+      - name: Install `uniffi-bindgen` for maturin
+        run: cargo install --path crates/iota-sdk-ffi --bin uniffi-bindgen
+
+      - name: Install Python
+        uses: actions/setup-python@v5
+        with:
+          python-version: 3.9
+
+      - name: Update version in pyproject.toml
```

Comment:
There is lots of duplicated code, maybe you can factorise with a better matrix. I factorised my workflow a lot after seeing Thoralf's matrix.

Reply:
I respectfully disagree, the workflow is currently very heavy to process because if the repetition and maintenance is going to be pretty annoying.

> on MacOS you have to use sed differently than on Linux: sed -i ... vs sed -i '' ...

Not an issue, we're already using a platform independent `sed - i` in scripts and CI, just need to tweak it a bit

> on Windows the default shell is PowerShell, so for platform independence I would have add shell: bash, which is redundant on Mac and Linux.

Doesn't sound like a problem to be

Reply:
So, I spent a little bit of time to find a compromise here, because IMHO conditional steps are also repetitive, which is why I factored out the repeated steps into an action and call it with different args instead.

Reply:
Okay, let me try to make an argument for this solution:

1. IMHO it scales a bit better than the `if` guarded steps, bc instead of adding a new step with a different `if`, repeating most of it and just passing some other variable, you essentially create callable functions (action.yml) with input arguments and return values (outputs). That makes it like any other function call where that function lives in another file (we do it everywhere in Rust)
2. `if` steps are not visually indented like regular code is, which makes it hard to quickly understand which parts run and which don't given a certain condition